src/main/java/com/ericsson/ei/notifications/HttpRequest.java

```diff
@@ -154,6 +157,14 @@ private void addAuthenticationData() throws AuthenticationException {
         String username = subscriptionField.get("userName");
         String password = subscriptionField.get("password");
 
+        if (!jasyptEncryptorPassword.isEmpty() && checkIfPasswordEncrypted(password)) {
+            TextFormatter textFormatter = new TextFormatter();
+            StandardPBEStringEncryptor encryptor = new StandardPBEStringEncryptor();
+            encryptor.setPassword(jasyptEncryptorPassword);
+            String encryptedPassword = textFormatter.removeEncryptionParentheses(password);
+            password = encryptor.decrypt(encryptedPassword);
```

Comment:
Is this the request for creating/updating subscription... shouldn't password be encrypted instead of decrypted?
You are now using "encryptor.decrypt()" method.

Reply:
This is the request when doing a notification. So I need to decrypt the password first and attach it as a header.

Reply:
ok